web/src/engine/main/src/headless/inputProcessor.ts

```diff
@@ -48,7 +48,11 @@ export class InputProcessor {
 
     this.contextDevice = device;
     this.kbdProcessor = new KeyboardProcessor(device, options);
-    this.lngProcessor = new LanguageProcessor(predictiveTextWorker, this.contextCache);
+    try {
+      this.lngProcessor = new LanguageProcessor(predictiveTextWorker, this.contextCache);
```

Comment:
This puzzles me, isn't the issue with Yeti that we cannot construct `predictiveTextWorker`, so wouldn't this go wrong (much) earlier?

Reply:
https://github.com/keymanapp/keyman/blob/263f918ea21ddaac1bc007a8ab8a672d21e74162/web/src/engine/predictive-text/worker-main/src/web/sourcemappedWorker.ts#L4-L10

The issue is that we cannot allow line 9 above to run - we can't instantiate the _actual_ `Worker`.  `predictiveTextWorker` above corresponds to an instance of the class `SourcemappedWorker` here.  

There's also a Node-friendly variant for use in headless testing, which is why the abstraction exists:

https://github.com/keymanapp/keyman/blob/263f918ea21ddaac1bc007a8ab8a672d21e74162/web/src/engine/predictive-text/worker-main/src/node/sourcemappedWorker.ts#L6-L15

`MappedWorker` there uses Node workers internally, but their spec differs enough from the Web worker type that we needed special handling for it.

Reply:
Looking at where `constructInstance()` is called, it seems to be called at the very top-level, e.g. 

https://github.com/keymanapp/keyman/blob/6591c61eeb82c88a8f26f511d20c5233dc9b1e91/web/src/app/browser/src/release-main.ts#L11-L16

So I am not sure how this fix handles that?

Reply:
Huh.  Makes me wonder why it seemed to work with the artificial repro attempt I tried.  🤷  Good point.